Makefile

```
@@ -55,8 +55,8 @@ repl:
 	dune utop src/haz3lcore
 
 test:
-	dune build @src/fmt @test/fmt --auto-promote src test --profile dev
-	node $(TEST_DIR)/haz3ltest.bc.js
+	dune fmt --auto-promote
+	dune build @src/fmt @test/fmt --auto-promote src test --profile dev; node $(TEST_DIR)/haz3ltest.bc.js
```

What prompted this change?

---

I'm mostly curious if we want dune fmt to be passive before build in all of the make commands or if there's some other workflow

---

Previously you would need to either run `make test` twice or `make fmt` then `make test` each time you wanted to run unit tests. This seemed to be due to the formatting causing the command to fail before the tests could be ran. This change seems to have fixed that and now runs the unit tests in the same command as formatting.

---

That makes sense. If we make this change for test we should probably make it for dev. @disconcision do you have any thoughts on that? Either way I think the last line can still be split onto two

```suggestion
	dune build @src/fmt @test/fmt --auto-promote src test --profile dev
	node $(TEST_DIR)/haz3ltest.bc.js
```

---

@green726 I tried addressing this here https://github.com/hazelgrove/hazel/pull/1379

Can you try those make commands out when you get a chance

---

I tried them out and approved the pull req - they worked. Thank you!